2018-11-02  Luke Hyun Jung  <luke@lnxsrv09.seas.ucla.edu>

	* lib/timespec.h (timespec_cmp):
	Use if statements to avoid using subtraction
	To not have integer overflow with large numbers away

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2018-11-02 22:58:04.758886000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec < b.tv_sec)
+    return -1;
+  if(a.tv_sec == b.tv_sec)
+    return 0;
+  if(a.tv_sec > b.tv_sec)
+    return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
